h/services/profile_group.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+
+
+class ProfileGroupService(object):
+
+    """A service for retrieving groups associated with users."""
+
+    def __init__(self, session, open_group_finder):
```

Comment:
What's the reason for injecting `open_group_finder` here instead of just hardcoding it?

Reply:
@seanh Sorry, hard-coding it where? Have patience with me :).

Reply:
I just mean that instead of `profile_groups_factory` doing `request.find_service(name='authority_group')` and then passing `auth_group_svc` into `ProfileGroupService()` as the `open_group_finder` argument, couldn't `ProfileGroupService.__init__()` just do `self.open_group_finder = request.find_service(name='authority_group')` itself? Does it need to be a param / are we ever going to want to be able to pass different `open_group_finder`s into different `ProfileGroupService`s? However, I guess this suggestion would just mean you'd have to pass `request` in to `ProfileGroupService` instead! So maybe it's silly. Happy to just leave it as it is!

Reply:
Ah! I was initially following the pattern I saw here: https://github.com/hypothesis/h/blob/7ae2cb90f4b74e8ef3b90d9de1b85d4b47c12990/h/services/group.py#L129

Two reasons; one is far more meaningful than the other. First, this keeps us from chucking the entire `request` into the service, and instead has us passing just the relevant bits.

The second reason (I warned you it was weak/indefensible) is that it's a lot easier to test this approach, as we don't have to cobble together a mock of a complex `request` for every test.

Reply:
Hmm, yeah, keep the request obj out of there is a pretty good reason I think